src/plugins/custom_integrations/common/index.ts

```diff
@@ -46,14 +46,19 @@ export const CATEGORY_DISPLAY = {
 
 export type Category = keyof typeof CATEGORY_DISPLAY;
 
+export interface CustomIntegrationIcon {
+  src: string;
+  type: 'eui' | 'svg';
```

Comment:
Agreed that the type should be more specific. Alternatively, we could use a type like this so that only the valid options are possible and we can access the `icon` and `path` fields without type guards:

```ts
type CustomIntegrationIcon =
  { type: 'eui'; icon: EuiIcon } |
  { type: 'path'; path: string; };
```

Reply:
Agreed on improvements. discussed offline with @clintandrewhall for path forward - we will address the icon-type refactor in a separate PR, since it addresses a parallel concern and this PR can be kept smaller. There's other type improvements that need to happen regardless (e.g. category-registration, fleet-card registration when https://github.com/elastic/package-registry/pull/735, ...), which we can address in isolation.